PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightsScalaProjectWizard.java

```diff
@@ -155,7 +160,7 @@ public boolean canFinish() {
 	@Override
 	public boolean performFinish() {
 		try {
-			CreateProjectUtil.createSampleFile(this.id, this.hdInsightScalaPageOne.getProjectName(), this.isUsingMaven, this.sparkVersion);
+			CreateProjectUtil.createSampleFile(this.id, this.hdInsightScalaPageOne.getProjectName(), this.isUsingMaven, this.sparkVersion.get());
```

Comment:
```
getSparkVersion()
    .ifPresent(sparkVer -> CreateProjectUtil.createSampleFile(this.id, this.hdInsightScalaPageOne.getProjectName(), this.isUsingMaven, sparkVer);

```

Otherwise, sparkVersion.get() will throw an exception when it's empty.

Reply:
Updated. Actually changed other place after the Optional change. For this one. It cannot be null since it is internal call so leave it there.

---
In reply to: [153125048](https://github.com/Microsoft/azure-tools-for-java/pull/1262#discussion_r153125048) [](ancestors = 153125048)